src/main/java/uk/gov/hmcts/reform/sscs/service/FuzzyMatcherService.java

```diff
@@ -1,103 +1,151 @@
+import static org.apache.commons.lang3.StringUtils.stripToEmpty;
 import static uk.gov.hmcts.reform.sscs.ccd.domain.Benefit.*;
-import static uk.gov.hmcts.reform.sscs.ccd.domain.Benefit.findBenefitByDescription;
 
+import com.google.common.collect.ImmutableSet;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Collectors;
+import java.util.Set;
 import java.util.stream.Stream;
 import lombok.extern.slf4j.Slf4j;
 import me.xdrop.fuzzywuzzy.FuzzySearch;
-import me.xdrop.fuzzywuzzy.model.ExtractedResult;
+import me.xdrop.fuzzywuzzy.model.BoundExtractedResult;
+import org.apache.commons.lang3.tuple.Pair;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.sscs.ccd.domain.Benefit;
 
 @Service
 @Slf4j
 public class FuzzyMatcherService {
-    private static final int THRESHOLD_SCORE = 90;
-
-    private static final Map<String, Benefit> FUZZY_WORD_MATCH_MAP = Map.of(
-        "personal", PIP,
-        "independence", PIP,
-        "universal", UC,
-        "employment", ESA
-    );
-
-    private static final Map<String, Benefit> EXACT_WORD_MATCH = Map.ofEntries(
-        Map.entry("AA", ATTENDANCE_ALLOWANCE),
-        Map.entry("IS", INCOME_SUPPORT),
-        Map.entry("liv", DLA),
-        Map.entry("SF", SOCIAL_FUND),
-        Map.entry("PC", PENSION_CREDITS),
-        Map.entry("RP", RETIREMENT_PENSION),
-        Map.entry("BB", BEREAVEMENT_BENEFIT),
-        Map.entry("MA", MATERNITY_ALLOWANCE),
-        Map.entry("IIDB", IIDB),
-        Map.entry("BSPS", BEREAVEMENT_SUPPORT_PAYMENT_SCHEME),
-        Map.entry("BSPA", BEREAVEMENT_SUPPORT_PAYMENT_SCHEME)
+    private static final int THRESHOLD_MATCH_SCORE = 90;
+    private static final int MAX_FUZZY_SEARCH_LENGTH = 4;
+
+    private static final List<String> EXACT_WORDS_THAT_WILL_NOT_CAUSE_A_MATCH = List.of(
+        "support",
+        "allowance",
+        "benefit",
+        "pension",
+        ""
     );
 
-    private final List<String> fuzzyChoices;
-
-    public FuzzyMatcherService() {
-        List<String> allMatches = stream(Benefit.values()).flatMap(benefit -> Stream.of(benefit.getShortName(), benefit.getDescription())).collect(Collectors.toList());
-        allMatches.addAll(FUZZY_WORD_MATCH_MAP.keySet());
-        fuzzyChoices = unmodifiableList(allMatches);
-    }
+    private static final Set<Pair<String, Benefit>> EXACT_WORDS_THAT_WILL_CAUSE_A_MATCH =
+        ImmutableSet.<Pair<String, Benefit>>builder()
+            .addAll(
+                Set.of(
+                    Pair.of("AA", ATTENDANCE_ALLOWANCE),
+                    Pair.of("IS", INCOME_SUPPORT),
+                    Pair.of("SF", SOCIAL_FUND),
+                    Pair.of("PC", PENSION_CREDITS),
+                    Pair.of("RP", RETIREMENT_PENSION),
+                    Pair.of("BB", BEREAVEMENT_BENEFIT),
+                    Pair.of("CA", CARERS_ALLOWANCE),
+                    Pair.of("Carers", CARERS_ALLOWANCE),
```

Comment:
How come we have Carers here for Carers Allowance but we don't have words like this for all the other benefit types?

Reply:
Removed. It's not needed here, since it can be fuzzy matched.